src/components/common/YourCompany/OfficeAddresses.vue

```diff
@@ -500,21 +503,22 @@ export default class OfficeAddresses extends Mixins(CommonMixin) {
    * Sets local address data and "inherit" flags from store.
    */
   private setLocalProperties (): void {
-    if (this.getOfficeAddresses?.[this.officeType]) {
```

Comment:
Using a "dynamic office type" seemed cool, but for filings/entities with registered and records addresses, the dynamic office type was the first one (registered) and then the code in this component had a special case for records office.

I found this pretty confusing and error-prone.

So I deleted "dynamic office type" and moved the address logic here. It's more explicit and, IMHO, much easier to understand the various scenarios.